Source/DataSourceProvider.swift

```diff
@@ -37,32 +37,34 @@ where CellFactory.Item == DataSource.Item, SupplementaryFactory.Item == DataSour
     public let supplementaryFactory: SupplementaryFactory
 
     fileprivate var bridgedDataSource: BridgedDataSource?
-    fileprivate var _tableEditingController: TableEditingController?
+    fileprivate var _tableEditingController: TableEditingController<DataSource>?
 
     // MARK: Initialization
 
     /**
      Initializes a new data source provider.
 
-     - parameter dataSource:           The data source.
-     - parameter cellFactory:          The cell factory.
-     - parameter supplementaryFactory: The supplementary view factory.
-
+     - parameter dataSource:             The data source.
+     - parameter cellFactory:            The cell factory.
+     - parameter supplementaryFactory:   The supplementary view factory.
+     - parameter tableEditingController: The tableEditing controller.
+     
      - returns: A new `DataSourceProvider` instance.
 
      - warning: Table views do not have supplementary views, but this parameter is still required in order to satisfy
      the generic constraints for Swift. You can simply pass the same `cellFactory` here. The parameter will be ignored.
      The same applies to collection views that do not have supplementary views. Again, the parameter will be ignored.
      */
-    public init(dataSource: DataSource, cellFactory: CellFactory, supplementaryFactory: SupplementaryFactory) {
+    public init(dataSource: DataSource, cellFactory: CellFactory, supplementaryFactory: SupplementaryFactory, tableEditingController: TableEditingController<DataSource>? = nil) {
```

Comment:
With the current update it is possible for the user to pass the `tableEditingController ` through the initialiser. However I have still left the old implementation with the `_tableEditingController`  for the time being. I was thinking of deprecating the old way (assigning it through the variable`tableEditingController`) but I wanted your opinion first @jessesquires .